src/librustc_passes/ast_validation.rs

```diff
@@ -864,6 +895,17 @@ impl<'a> Visitor<'a> for AstValidator<'a> {
                              "`async fn` is not permitted in the 2015 edition").emit();
         }
     }
+
+    fn visit_impl_item(&mut self, ii: &'a ImplItem) {
+        match ii.node {
+            ImplItemKind::Method(ref sig, ref body) => {
+                self.check_fn_decl(&sig.decl);
+                self.visit_block(body);
```

Comment:
This seems unnecessary, the block should be already visited by the `walk_impl_item` below.